packages/overlay/src/integrations/sentry/components/traces/spans/SpanTree.tsx

```diff
@@ -21,10 +23,29 @@ export default function SpanTree({
   spanNodeWidth: number;
   setSpanNodeWidth?: (val: number) => void;
 }) {
+  const { query, matchesQuery, showOnlyMatched } = useSearch();
+
+  const filteredTree = useMemo(() => {
+    if (!query) return tree;
+    if (showOnlyMatched) {
+      const spanMemo = new Map<string, boolean>();
```

Comment:
This implementation is not bad but I don't think the cache would be shared across children/siblings. I think this cache should be moved to `useSearch()` level and get reset when `query` changes.